lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -46,14 +46,22 @@ def call(env)
           # configure the Rack middleware once
           configure()
 
-          # start a new request span and attach it to the current Rack environment;
-          # we must ensure that the span `resource` is set later
-          request_span = @tracer.trace(
-            'rack.request',
+          trace_options = {
             service: @service,
             resource: nil,
             span_type: Datadog::Ext::HTTP::TYPE
-          )
+          }
+
+          # Merge distributed trace ids if present
+          unless env['HTTP_X_DDTRACE_PARENT_TRACE_ID'].nil? || env['HTTP_X_DDTRACE_PARENT_SPAN_ID'].nil?
```

Comment:
@ejholmes indeed there are still different values around but as I said just above, plan is really to have all of them aligned. There has been recent moves on Python recently concerning this: https://github.com/DataDog/dd-trace-py/blob/c609a9054c479b6e394eb807520567b000f1aace/ddtrace/contrib/aiohttp/middlewares.py#L12 and in Go too: https://github.com/DataDog/dd-trace-go/blob/f3e02e97f8a21336bac49d9071d27edd98885da3/tracer/contrib/tracegrpc/grpc.go#L17 and Ruby, with the help of this patch, should follow.